src/backend/distributed/utils/multi_transaction.c

```diff
@@ -347,3 +348,62 @@ ConnectionList(HTAB *connectionHash)
 
 	return connectionList;
 }
+
+
+/*
+ * OpenConnectionsToShardPlacements opens connections to all placements of the
+ * shard with the given shardId and populates the shardConnectionHash table
+ * accordingly.
+ */
+void
+OpenConnectionsToShardPlacements(uint64 shardId, HTAB *shardConnectionHash,
```

Comment:
The other functions like GetShardConnections are also candidates to go into that file.